src/common/code_utils.hpp

```diff
@@ -104,4 +104,10 @@
         goto exit;   \
     } while (false)
 
+/**
+ * This flag indicates the parameter may be unused.
+ *
+ */
+#define OTBR_UNUSED __attribute__((unused))
```

Comment:
Is there a goal to support a wide range of toolchains (e.g. other than clang/gcc)?

Reply:
I tend to add the support when it is needed. Otherwise it may increase maintenance burden, thoughts?

Reply:
Are we confident that adding `OTBR_UNUSED` macro to each declaration is sufficient? OpenThread, LwIP, etc. went with a different model (i.e. after declaration) primarily due to compiler support issues.

Reply:
changed to reuse OT_UNUSED_VARIABLE